src/java.desktop/share/classes/sun/awt/SunToolkit.java

```diff
@@ -1442,21 +1442,21 @@ public void realSync() {
      * sub-system, flushing all pending work and waiting for all the
      * events to be processed.  This method guarantees that after
      * return no additional Java events will be generated, unless
-     * cause by user. Obviously, the method cannot be used on the
-     * event dispatch thread (EDT). In case it nevertheless gets
+     * caused by user. Obviously, the method cannot be used on the
+     * event dispatch thread (EDT). In case it gets
      * invoked on this thread, the method throws the
-     * IllegalThreadException runtime exception.
+     * {@code IllegalThreadException} runtime exception.
      *
-     * <p> This method allows to write tests without explicit timeouts
-     * or wait for some event.  Example:
+     * <p> This method allows writing tests without explicit timeouts
+     * or waiting for some event.  Example:
```

Comment:
Should it be “without _using_ … or waiting …”?  
Does “for an event” sound better?

Reply:
In my opinion, "without using ... or waiting ..." seems better to match the tense of "waiting" to show more clearly what the "or" is intended for (assuming the OR is meant for the explicit timeouts and the waiting for an event).

Otherwise, the "or" may be confused to be between "writing ... or waiting ...".

For the "event" portion, I think either here is OK. I don't think either would cause confusion and both sound reasonable.